docs/proposals-accepted/202107-planner-component.md

```diff
@@ -0,0 +1,73 @@
+## Separate Planner into its own Component
+
+* **Owners:**
+
+    * [@andrejbranch](https://github.com/andrejbranch)
+
+* **Related Tickets:**
+
+    * [Compact all available plans for a single group concurrently](https://github.com/thanos-io/thanos/issues/3806)
+
+## Summary
+
+Separate compactor's planner into its own component and have compactor workers request available plans from the planner.
+
+## Why
+
+Currently, the Thanos compactor handles planning which blocks to compact as well as vertical/horizontal compaction of the groups. This causes issues with the scalability of compaction for a single tenant. The actual compaction process is relatively simple, as all the process needs to know is what blocks to compact. By contrast, the planning logic is relatively complex and currently can't coordinate with other compactors. To get around the coordination issue, currently the compactor can be sharded by tenant. This will not horizontally scale. When a single tenant becomes very large it will need to distribute the workload across many compaction processes across many instances, regardless of how the workload is divided.
+
+## Goals
+
+* To separate planner into its own component and have compactors communicate with the planner to get the next available plan.
+
+## Non-Goals
+
+* Horizontally scaling compaction for a single tenant will be addressed following the completion of this proposal
+
+## Proposal
```

Comment:
Can we have some `### Risks` section where we can explain what we do to ensure Planner is not Single point of failure? (HA / Scalability)